bat/workload.go

```diff
@@ -172,6 +152,35 @@ func CreatePublicWorkload(ctx context.Context, tenant string, opt WorkloadOption
 	return createWorkload(ctx, "", opt, config, true)
 }
 
+// CreateWorkloadFromFile will call ciao-cli workload create -yaml
+// to create a workload from the specified file.  ciao-cli will be invoked
+// using the admin user if the public flag is set to true.  The provided
+// workload definition file will remain intact after the call returns.
+// The id of the new workload is returned upon success.
+func CreateWorkloadFromFile(ctx context.Context, public bool, tenant, wdPath string) (string, error) {
+	args := []string{"workload", "create", "-yaml", wdPath}
+
+	var data []byte
+	var err error
+	if public {
+		data, err = RunCIAOCLIAsAdmin(ctx, tenant, args)
+	} else {
+		data, err = RunCIAOCLI(ctx, tenant, args)
+	}
+
+	if err != nil {
+		return "", err
```

Comment:
The error message could be better but it shouldn't be fixed in this PR.  This is not new code, it is existing code that has just be refactored into a function.  Also, the bat package doesn't use the errors package yet.